src/Akka.Hosting/AkkaConfigurationBuilder.cs

```diff
@@ -94,6 +94,13 @@ public sealed class AkkaConfigurationBuilder
         /// Used to hold a reference to the <see cref="Sys"/> being started.
         /// </summary>
         internal Option<ActorSystem> Sys { get; set; } = Option<ActorSystem>.None;
+        
+        /// <summary>
+        /// INTERNAL API.
+        ///
+        /// Used to hold a reference to the dependency injection <see cref="ServiceProvider"/> being used to setup the <see cref="ActorSystem"/>.
+        /// </summary>
+        internal Option<IServiceProvider> ServiceProvider { get; set; } = Option<IServiceProvider>.None;
```

Comment:
We don't need to hold onto this here - it will be passed back into us in ` ActorSystemFactory()`

Reply:
Ww actually do, this will be used by extension methods to pull required services from the builder class

Reply:
This is pulled when we create the `LoggerSetup` right?

Reply:
Because if that's the case, no we don't - look at how we create the `DependencyResolverSetup` for instance:

https://github.com/akkadotnet/Akka.Hosting/blob/2f1683dd42e237badd823bb2ff4739d5efe51020/src/Akka.Hosting/AkkaConfigurationBuilder.cs#L227-L239

Reply:
The ILoggerFactory needs the service provider so that it can do an ILoggerFactory lookup _before_ the actor system starts, extension methods only have access to the builder instance, this is the only generic way I can think of for doing that, instead of trying to do a hook inside the actor system factory itself

Reply:
another way of doing this is to ask the user to provide the service provider instance for the ILoggerFactory logger extension method, which is the thing i'm trying hard to avoid

Reply:
The way the ILoggerFactory logger cheats the DI system is to do a lookup before the actor system starts, put it inside a Setup class, and pulls that Setup class in its constructor.
The big problem is that there is no way to access the ServiceProvider instance from inside an extension method, hence we have to provide that inside the builder instance itself

Reply:
I fixed this in a subsequent commit. Didn't need to change any of the public APIs.